common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -1175,7 +1176,10 @@ def get_demand_hint(self, hint_index):
                     )
                 ),
                 # Course-authored HTML demand hints are supported.
-                hint_text=HTML(self.runtime.replace_urls(get_inner_html_from_xpath(demand_hints[counter])))
+                hint_text=HTML(self.runtime.service(self, "replace_urls").replace_urls(
+                    get_inner_html_from_xpath(demand_hints[counter]),
+                    static_replace_only=True
+                ))
```

Comment:
`static_replace_url` is also used in [studio preview](https://github.com/open-craft/edx-platform/blob/bbeccd8232d4d83ee758aa0e4620f0d43b2e668b/cms/djangoapps/contentstore/views/preview.py#L179)

I introduced this argument, only to keep the logic consistent in these two places before and after this PR, where only the static urls were being replaced and not the course or jump-to-id urls.

In case of hints, I think replacing course or jump-to-id urls in the hints section could be useful in a scenario where the course author may want to provide a link to a portion of a course content in the assessment hints.

In the case of [studio preview](https://github.com/open-craft/edx-platform/blob/bbeccd8232d4d83ee758aa0e4620f0d43b2e668b/cms/djangoapps/contentstore/views/preview.py#L179), I am not sure about the reasoning behind not replacing course urls (not replacing jump-to-id url make sense as the course is dynamic in the studio and the ids may change). Do you think, me should replace the course urls here as well ? In that case we can get rid of the `static_replace_url` argument.

Reply:
@kaustavb12,

> static_replace_url is also used in [studio preview](https://github.com/open-craft/edx-platform/blob/bbeccd8232d4d83ee758aa0e4620f0d43b2e668b/cms/djangoapps/contentstore/views/preview.py#L179)

Ah, that's right.

> Do you think, we should replace the course urls here as well ? In that case we can get rid of the static_replace_url argument.

Right now Studio is just providing the `{STUDIO_URL}/course/about` URL. After this change, it would be pointing to `{STUDIO_URL}/courses/{COURSE_ID/about`. These pages do not exist in Studio anyway, so this won't make a too big difference. Let's leave it as-is, for consistency with the existing approach.

> not replacing jump-to-id url make sense as the course is dynamic in the studio and the ids may change

It could be nice to be able to check within Studio that the ID is correct by navigating there. However, the implementation would probably need to change for the Studio preview (the URLs are different), so let's omit this.